api/db/populate-friendships.js

```diff
@@ -0,0 +1,85 @@
+const { getRandomSubset, getRandomDate, createPrompt } = require('./utils');
+
+const User = require('../models/User');
+const Friendship = require('../models/Friendship');
+
+let displayLogs;
+
+const createFriendship = async (friendshipParams) => {
+  const friendship = new Friendship({
+    requestant: friendshipParams.requestant,
+    receivant: friendshipParams.receivant,
+    accepted: friendshipParams.accepted,
+    acceptedAt: friendshipParams.acceptedAt,
+    lastMessageAt: friendshipParams.lastMessageAt,
+  });
+  await friendship.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New friendship created: ${friendship}`);
+  }
+  return friendship;
+};
+
+const createFriendships = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating friendships collection...');
+  displayLogs = verbose;
+
+  let usersList = await User.distinct('_id');
+  const friendshipsArray = await Friendship.find();
+  let recordCount = friendshipsArray.length;
+
+  let proceed = true;
+  if (recordCount > 0) {
+    proceed = await createPrompt(
+      'Some users already have friends. Would you like to add friends to users that already contain friends?'
+    );
+  }
+
+  for (const user of usersList) {
+    const containFriends = friendshipsArray.some(
+      (friendship) =>
+        friendship.requestant.toString() === user.toString() ||
+        friendship.receivant.toString() === user.toString()
+    );
+    if (!proceed && containFriends) continue;
+
+    // split current user from all others
+    const tmpList = usersList.filter((userValue) => userValue != user);
+    // create subset userList from remaining usersList
+    const usersSubset = getRandomSubset(tmpList, tmpList.length);
+
+    for (const receivant of usersSubset) {
+      const alreadFriends = friendshipsArray.filter((friendship) => {
```

Comment:
I think this should be a `some` instead of a `filter`. Also, you can remove the `{ return (  ) }` part and only leave everything in between. Oh, and a nitpick: you've missed the `y` of `already`.